### ISSUE TITLE
Fix Request: false positive in FieldOfHashCodeMissingInEquals

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#349]
```
class Amount {

    private BigDecimal amountValue;
    private String amountCurrency;

    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        Amount other = (Amount) o;
        // avoid BigDecimal equals which compares value and also scale, use compareTo instead
        return amountValue.compareTo(other.amountValue) == 0 && amountCurrency.equals(other.amountCurrency);
    }
    public int hashCode() {
        int result = null != amountValue ? amountValue.hashCode() : 0;
        result = 31 * result + amountCurrency.hashCode();
        return result;
    }
}
```

**Comment by jborgers** [comment]
This is fixed with the pmd7 version of the rule
